controller/src/test/java/org/jboss/as/controller/EnvVarAttributeOverrideModelTestCase.java

```diff
@@ -0,0 +1,150 @@
+import org.jboss.as.controller.test.AbstractControllerTestBase;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.junit.Test;
+
+/**
+ * @author jmesnil
+ */
+public class EnvVarAttributeOverrideModelTestCase extends AbstractControllerTestBase {
+    private static final SimpleAttributeDefinition MY_ATTR = new SimpleAttributeDefinitionBuilder("my-attr", ModelType.STRING, true)
+            .setAllowExpression(true)
+            .build();
+
+    private static PathAddress CUSTOM_RESOURCE_ADDR = PathAddress.pathAddress("subsystem", "custom");
+
+    @Override
+    protected void initModel(ManagementModel managementModel) {
+        ManagementResourceRegistration rootRegistration = managementModel.getRootResourceRegistration();
+        // register the global operations to be able to call :read-attribute and :write-attribute
+        GlobalOperationHandlers.registerGlobalOperations(rootRegistration, processType);
+        // register the global notifications so there is no warning that emitted notifications are not described by the resource.
+        GlobalNotifications.registerGlobalNotifications(rootRegistration, processType);
+
+        ResourceDefinition profileDefinition = createDummyProfileResourceDefinition();
+        rootRegistration.registerSubModel(profileDefinition);
+    }
+
+    private static ResourceDefinition createDummyProfileResourceDefinition() {
+        return ResourceBuilder.Factory.create(CUSTOM_RESOURCE_ADDR.getElement(0),
+                new NonResolvingResourceDescriptionResolver())
+                .setAddOperation(new AbstractAddStepHandler(Arrays.asList(MY_ATTR)))
+                .setRemoveOperation(ReloadRequiredRemoveStepHandler.INSTANCE)
+                .addReadWriteAttribute(MY_ATTR, null, new ReloadRequiredWriteAttributeHandler(MY_ATTR))
+                .build();
+    }
+
+    private static boolean isFeatureEnabled() {
+        return EnvVarAttributeOverrider.isEnabled();
+    }
+    /**
+     * This test creates a resource with an attribute set to "value1".
+     * The test runs with an env var corresponding to the resource attribute set to "value2"
+     *
+     * When we read the attribute value we verify that the value comes from the env var (i.e. "value2").
+     *
+     * @throws OperationFailedException
+     */
+    @Test
+    public void testOverriddenSpecifiedAttributeValue() throws OperationFailedException {
+        ModelNode addOp = createOperation("add", CUSTOM_RESOURCE_ADDR);
+        addOp.get(MY_ATTR.getName()).set("value1");
+        executeCheckNoFailure(addOp);
+
+        ModelNode readResource = createOperation(READ_RESOURCE_OPERATION, CUSTOM_RESOURCE_ADDR);
+        executeCheckNoFailure(readResource);
+
+        ModelNode readAttribute = createOperation(READ_ATTRIBUTE_OPERATION, CUSTOM_RESOURCE_ADDR);
+        readAttribute.get(NAME).set(MY_ATTR.getName());
+        ModelNode response = executeCheckNoFailure(readAttribute);
+        if (isFeatureEnabled()) {
```

Comment:
Is this test somehow running automatically for both `true` and `false` branches, or is this designed so that you have to manually change/comment out `controller/pom.xml#186` and then run the testsuite again?

Reply:
I believe that even if the enabling functionality was broken and the feature was always on, this would still be green with the feature "disabled", as there is nothing trying to rewrite "value1" to "value2" in `default-test` execution.
Would it be possible to set 
```
                            <environmentVariables>
                                <SUBSYSTEM_CUSTOM__MY_ATTR>value2</SUBSYSTEM_CUSTOM__MY_ATTR>
                            </environmentVariables>
```
in execution `default-test`? Than we would be sure that the on/off switch is working as intended.

Reply:
@vsalbaba good catch, I've updated the PR based on your suggestion